draft-birkholz-cose-cometre-ccf-profile.md

```diff
@@ -38,26 +38,125 @@ author:
+which addresses the persistance of committed transactions that happen between new signatures of
+the Merkle Tree root.
 
 ## Requirements Notation
 
 {::boilerplate bcp14-tagged}
 
-# CoMETRE Confidential Consortium Framework Profile
+# Description of the CCF Tree Algorithm
+
+Recall the definition of CoMeTre inclusion proofs, which are parametrized by 3 CBOR data types that are specific to the Tree Algorithm:
+
+~~~~ cddl
+signed-inclusion-proof = [
+  signed-inclusion-proof: bstr .cbor smtr ; the payload is a merkle root, as described by the tree algorithm, and is detached.
+  inclusion-proof: bstr .cbor CCF-inclusion-proof; the inclusion-proof, as described in the tree algorithm
+  leaf: bstr .cbor CCF-leaf ; the leaf, as described in the tree algorithm
+]
+~~~~
+
+This document defines the `CCF-feaf` and `CCF-inclusion-proof` CBOR types. The signed Merkle Root data type `smtr` is the same as in {{-COMTRE}} but MUST set the protected header parameter carrying the identifier of the tree algorithm, `tree_alg`, to the value TBD_1.
+
+## Tree Shape
+
+The input of the Merkle Tree Hash (MTH) function is a list of n bytestrings, written D_n = \{d\[0\], d\[1\], ..., d\[n-1\]\}. The output is a single HASH_SIZE bytestring, also called the tree root hash.
+
+This function is defined as follows:
+
+The hash of an empty list is the hash of an empty string:
+
+~~~
+MTH({}) = HASH().
+~~~
+
+The hash of a list with one entry (also known as a leaf hash) is:
+
+~~~
+MTH({d[0]}) = HASH(d[0]).
+~~~
+
+For n > 1, let k be the largest power of two smaller than n (i.e., k < n <= 2k). The Merkle Tree Hash of an n-element list D_n is then defined recursively as:
+
+~~~
+MTH(D_n) = HASH(MTH(D[0:k]) || MTH(D[k:n])),
+~~~
+
+where:
+
+- \|\| denotes concatenation
+- : denotes concatenation of lists
+- D\[k1:k2\] = D'_(k2-k1) denotes the list \{d'\[0\] = d\[k1\], d'\[1\] = d\[k1+1\], ..., d'\[k2-k1-1\] = d\[k2-1\]\} of length (k2 - k1).
+
+## Leaf Components
 
-Context
+Each leaf in a CCF Merkle Tree carries the following components:
+
+~~~
+CCF-leaf = [
+  internal-hash: bstr ; a string of HASH_SIZE bytes;
+  internal-data: bstr; a string of at most 1024 bytes; and
+  data_hash: bstr ; the serialization of the element stored at this leaf.
+]
+~~~
+
+The `internal_hash` and `internal_data` bytestrings are internal to the CCF implementation. Similarly, the auxiliary ledger entries are internal to CCF. They are opaque to receipt Verifiers, but they commit the TS to the whole ledger contents and may be used for additional, CCF-specific auditing.
+
+## Signed Inclusion Proof Format
+
+CCF inclusion proofs consist of a list of digests tagged with a single left-or-right bit.
+
+~~~
```

Comment:
Maybe highlight again where this CDDL fragment is used? At the moment this is a little bit free-floating, maybe?

Reply:
Thanks, this was actually an inconsistency, the type is mentioned first when recalling the `signed-inclusion-proof` type, but it called `CCF-inclusion-proof` there. I fixed the inconsistency in 085fdd8 and also added a mention about the element index not being present